web/app/models.py

```diff
@@ -274,25 +274,50 @@ def update_current_print(self, filename, current_print_ts):
 
                 return
             LOGGER.warn(f'Print not properly ended before next start. Stale print_id: {self.current_print_id} - printer_id: {self.id}')
-            self.unset_current_print()
+            self.unset_current_print(gone=True)
             self.set_current_print(filename, current_print_ts)
         else:
             self.set_current_print(filename, current_print_ts)
 
-    def unset_current_print(self):
+    def unset_current_print(self, gone=False, failed=False, canceled=False):
         print = self.current_print
         self.current_print = None
         self.save()
 
         self.printerprediction.reset_for_new_print()
 
+        assert print
+
         if print.cancelled_at is None:
             print.finished_at = timezone.now()
             print.save()
 
         PrintEvent.create(print, PrintEvent.ENDED)
         self.send_should_watch_status()
 
+        event_name = 'PrintDone'
```

Comment:
Isn't this duplicate with https://github.com/TheSpaghettiDetective/TheSpaghettiDetective/pull/604/files#diff-f821bd2c29b6c78c53566d64c0e903b4875b3f53e6b3243da0ea75d0fc4bb2e4R46 ? I'm having a hard time to follow when these notifications are sent as it seems we have at least 3 places to send them.